doc/vrf/no-default-vrf-route-fallback-hld.md

```diff
@@ -0,0 +1,334 @@
+- The knob is intended only for backward compatibility and will be deprecated.
+
+> **Note:**  
+> The unreachable default route added to non-default VRFs exists only in the Linux kernel routing table.  
+> **It will not be added to APP_DB and will not be propagated to lower layers such as ASIC_DB, SAI, or programmed into hardware.**  
+> This ensures there is no impact on hardware FIB or ASIC resources.
+
+## 5. Requirements
+
+### 5.1. Functional Requirements
+
+- **FR-1**: On non-default VRF creation, add unreachable default route with high metric (for both IPv4 and IPv6).
+- **FR-2**: Provide a global CLI/DB knob to enable/disable kernel VRF fallback.
+- **FR-3**: When knob is **enabled**, remove unreachable default route from all non-default VRFs.
+- **FR-4**: When knob is **unset** or **removed** (default), restore unreachable default route to all non-default VRFs.
+- **FR-5**: CLI and show commands updated to reflect new behavior and knob state.
+- **FR-6**: Configuration persists across reboots and config reloads.
+- **FR-7**: Knob is deprecated and will be removed in a future release.
+
+### 5.2. Non-Functional Requirements
+
+- **NFR-1**: No impact on dataplane performance.
+- **NFR-2**: Minimal memory and CPU overhead.
+- **NFR-3**: No ASIC table consumption for unreachable routes.
+
+### 5.3. Exemptions/Limitations
+
+- **EX-1**: No change to kernel routing implementation (uses existing unreachable route mechanism).
+- **EX-2**: Knob is global only; no per-VRF override.
+
+## 6. Architecture Design
+
+- **CONFIG_DB**: Stores global fallback knob status.
+- **VRF Manager (vrfmgrd)**: Handles adding/removing unreachable default routes to non-default VRFs based on knob.
+- **sonic-utilities**: CLI and show command updates.
+- **Linux Kernel**: Manages unreachable route entries.
+
+```
+┌─────────────┐    ┌──────────────┐    ┌─────────────┐
+│             │    │              │    │             │
+│   CLI / GUI │───▶│  CONFIG_DB   │───▶│   vrfmgrd   │
+└─────────────┘    └──────────────┘    └─────────────┘
+                                          │
+                                          ▼
+                                 ┌───────────────────┐
+                                 │   Linux Kernel    │
+                                 │ (unreachable rt)  │
+                                 └───────────────────┘
+```
+
+## 7. High-Level Design
+
+### 7.1. Default Behavior
+
+- On creation of any non-default VRF, vrfmgrd adds:
+  - `ip route add unreachable default metric 4278198272 vrf <VRF_NAME>`
+  - (For both IPv4 and IPv6.)
+- This disables fallback in the kernel.
+- **This unreachable default route will not be added to APP_DB and will not be propagated to lower layers.**
+
+### 7.2. Global Fallback Knob
+
+- New global configuration (in CONFIG_DB and CLI): `KERNEL_VRF_FALLBACK`
```

Comment:
IIRC, in the HLD review meeting, we agreed to implement only the CONFIG_DB mechanism and not the CLI.